javav2/example_code/iotsitewise/src/test/java/SitewiseTests.java

```diff
@@ -0,0 +1,260 @@
+
+
+import com.example.iotsitewise.HelloSitewise;
+import com.example.iotsitewise.scenario.CloudFormationHelper;
+import com.example.iotsitewise.scenario.SitewiseActions;
+import com.google.gson.Gson;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.iotsitewise.model.CreateAssetModelResponse;
+import software.amazon.awssdk.services.iotsitewise.model.CreateAssetResponse;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class SitewiseTests {
+
+    private static final String assetModelName = "MyAssetModel"+ UUID.randomUUID();
+    private static final String assetName = "MyAsset";
+
+    private static String assetId = "";
+    private static final String portalName = "MyPortal";
+    private static String contactEmail = "";
+    private static final String gatewayName = "myGateway"+ UUID.randomUUID();
+    private static final String myThing = "myThing"+ UUID.randomUUID();
+
+    private static String assetModelHello = "";
+
+    private static String assetModelId = "";
+
+    private static String iamRole = "";
+
+    private static final SitewiseActions sitewiseActions = new SitewiseActions();
+
+    private static Map<String, String> propertyIds;
+
+    private static String humPropId = "";
+
+    private static String tempPropId = "";
+
+    private static String portalId = "";
+
+    private static String gatewayId = "";
+    private static final String ROLES_STACK = "RoleSitewise";
+
+    @BeforeAll
+    public static void setUp() {
+        CloudFormationHelper.deployCloudFormationStack(ROLES_STACK);
+        Map<String, String> stackOutputs = CloudFormationHelper.getStackOutputs(ROLES_STACK);
+        iamRole = stackOutputs.get("SitewiseRoleArn");
+
+        Gson gson = new Gson();
+        String json = getSecretValues();
+        SecretValues values = gson.fromJson(json, SecretValues.class);
+        contactEmail = values.getContactEmail();
+        assetModelHello  = values.getAssetModelHello();
```

Comment:
This will only work for you since it's in your account. Better to create the model as part of the class, or as I said in the Hello file, use a different action method.

Reply:
Lets discuss